### ISSUE TITLE
Failed to run Benchmarksql 5.0

### ISSUE THREAD

**Issue by ricemaster** [yugabyte/yugabyte-db#4485]
Hi,

I'm trying to run [BenchmarkSQL-5.0](https://sourceforge.net/projects/benchmarksql/) on yugabyte-db v2.1.6.0 which is deployed as a 3-node universe with 3 master and 3 tserver on my local server (Centos 7).

I did modified a few sql scripts in benchmark5.0 to make table-creating and  data-loading successfully. Such as I moved the `primary key` parts from `indexCreates.sql` into `tableCreates.sql` for yb doesn't support using `alter table` to add primary keys for now.

So all the tables, indexes, foreign keys are created by benchmarksql 5.0 and all the data loaded with no error.

But when I run Benchmarksql, the results shows lots of errors as below.
```
org.postgresql.util.PSQLException: ERROR: 
Operation failed. Try again.: ccad7050-0fbf-452a-845c-897bbf8fec58 
Conflicts with higher priority transaction: f9e056ba-c6ee-4a46-9d0c-d2ca3cefb334
```
or
```
org.postgresql.util.PSQLException: ERROR: 
Operation failed. Try again.: 
Value write after transaction start: { physical: 1589636243373238 logical: 4 } >= { physical: 1589636243366968 }
```

![2](https://user-images.githubusercontent.com/10449002/82121363-f8161a80-97be-11ea-9bb4-6fb9cb9a9917.png)

The yugabyte-db is using all default settings.

Is there any configurations I should do before run BenchmarkSQL?

Thanks.

**Comment by kmuthukk** [comment]
@ddorian , @psudheer21 -- can you take an initial look?

**Comment by ddorian** [comment]
The first one is about aborting conflicting transactions. In this case we return a Postgresql error code **`40001`** which means it failed and needs to retry the transaction.

The second one is the same, [explained here](https://github.com/yugabyte/yugabyte-db/issues/2706#issuecomment-550502629), the transaction needs to be retried.

I think the tool isn't retrying transactions when it gets the error code from Postgresql.

Looking further the last release date is 2016.

We maintained benchmarking tools:
1. TPCC https://github.com/yugabyte/tpcc
2. sysbench https://github.com/yugabyte/sysbench
3. YCSB https://github.com/yugabyte/YCSB

**Comment by ricemaster** [comment]
Hi @ddorian ，thank you so much for your reply.

We use BenchmarkSQL because we're trying to use one same tool to benchmark defferent db products.

And yes, I tried TPCC and YCSB, too.

In TPCC, it showed the results like
| Transaction Type Index | Transaction Name | Start Time (microseconds) | Latency (microseconds) | Worker Id (start number) | Phase Id (index in config file) |
| ---- | ---- | ---- | ---- | ---- | ---- |
| 1    | New Order | 1589511159 | 481895 | 1    | 0    |
| 1    | New Order | 1589511159 | 517238 | 0    | 0    |

and

|time(sec)| throughput(req/sec)| avg_lat(ms)| min_lat(ms)| 25th_lat(ms)| median_lat(ms)| 75th_lat(ms)| 90th_lat(ms)| 95th_lat(ms)| 99th_lat(ms)| max_lat(ms)| tp (req/s) scaled|
|-|-|-|-|-|-|-|-|-|-|-|-|
|0|0.400|499.567|481.895|481.895|517.238|517.238|517.238|517.238|517.238|517.238|0.002|

But which I need is a `tpmC` score in order to compare with other database running benchmarksql.

I do know that's because I'm not familiar with this tool, so it would be super appropriated if I may have your help to get a `tpmC` using TPCC.

Besides, when I use YCSB, I got the same error `Already present: Duplicate request` both in YCSB and BenchmarkSQL. I attached the screenshots below.

- YCSB
![d4fdb731e331cb12b307419c06f31e1](https://user-images.githubusercontent.com/10449002/82275921-78eb3700-99b6-11ea-82da-ac0fdd2cb69b.png)
![1cbb01d91c04d32d4eaca7a65a000aa](https://user-images.githubusercontent.com/10449002/82275929-7c7ebe00-99b6-11ea-8a0a-b476df4671eb.png)

- BenchmarkSQL
![49f8b8fd4ad2738442305561bb7015a](https://user-images.githubusercontent.com/10449002/82276155-0af33f80-99b7-11ea-9f99-5ad80ac5c86a.png)

Is that because I use wrong configuration?
I'm very glad to provide more details, but I don't know what kind of information would be useful. So if any log/commandline/parameter needed, please let me know.

**Comment by rkarthik007** [comment]
@psudheer21 @ndeodhar - could you folks please chime in here. Questions about TPCC.

@ricemaster - note that we're actively working on getting TPCC benchmarks running on YugabyteDB along with getting a tpmC number. But happy to take a look.

**Comment by ddorian** [comment]
@ricemaster we just released version [2.2 with new TPC-C benchmarks](https://blog.yugabyte.com/announcing-yugabytedb-2-2-distributed-sql-made-easy/) . Also [docs](https://docs.yugabyte.com/latest/benchmark/tpcc-ysql/) on how to run them.

**Comment by ricemaster** [comment]
> @ricemaster we just released version [2.2 with new TPC-C benchmarks](https://blog.yugabyte.com/announcing-yugabytedb-2-2-distributed-sql-made-easy/) . Also [docs](https://docs.yugabyte.com/latest/benchmark/tpcc-ysql/) on how to run them.

Thank you @ddorian so much for sharing these wonderful infomation. I'll try to run them on my own universe.
